cli/glnexus_cli.cc

```diff
@@ -674,8 +683,7 @@ int main_discover_alleles(int argc, char *argv[]) {
 
     // Write the discovered alleles to stdout
     YAML::Emitter yaml;
-    H("write results as yaml",
-      utils::yaml_of_contigs_alleles_ranges(contigs, ranges, valleles, yaml));
+    H("write results as yaml",utils::yaml_of_contigs_alleles(contigs, valleles, yaml));
```

Comment:
`valleles` can be collapsed into a single `discovered_alleles` structure before writing out to YAML (given that the ranges are non-overlapping, as you're now checking). The YAML file should then have one flat list of alleles, and deserialize into one `discovered_alleles`, NOT a `vector<discovered_alleles>`.